proto/iotics/api/advanced_search.proto

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2019-2022 Iotic Labs Ltd. All rights reserved.
+
+// Iotics Web protocol definitions (advanced search)
+syntax = "proto3";
+
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/timestamp.proto";
+import "google/rpc/status.proto";
+import "iotics/api/common.proto";
+import "iotics/api/feed.proto";
+import "iotics/api/input.proto";
+import "iotics/api/search.proto";
+package iotics.api;
+
+option csharp_namespace = "Iotics.Api";
+option go_package = "github.com/Iotic-Labs/iotic-go-proto-qapi/iotics/api;ioticsapi";
+option java_multiple_files = true;
+option java_outer_classname = "SearchProto";
+option java_package = "com.iotics.api";
+option objc_class_prefix = "IAX";
+option php_namespace = "Iotics\\Api";
+
+// ---------------------------------------------------------------------------------------------------------------------
+
+
+// AdvancedSearchAPI provides a set of RPCs to run search with advanced filtering (compared to SearchAPI).
+service AdvancedSearchAPI {
+
+  // TODO - this will be supported in future but is out of scope for initial requirements
+  // Run an advanced search request on a user timeout and return formatted results. Note that more than one result can
+  // be returned from each host and a final empty response indicates that no more results are expected from the
+  // associated host. (local and remote)
+  rpc AdvancedSearch(AdvancedSearchRequest) returns (stream SearchResponse) {}
```

Comment:
Re-use an object from another service. Why not add AdvancedSearch (which clearly has a different name) to the existing search service?

Reply:
I have seen this comment https://github.com/Iotic-Labs/api/pull/23#issuecomment-1220669866 but I'm not sure to get what is the problem if the name is clearly different

Reply:
>  Why not add AdvancedSearch (which clearly has a different name) to the existing search service?

The first commit (now c72bc656c7ac1a6457) shows that.
I think the only Q might be: Is "advanced search" the right name for this? (Especially if we end up deprecating the old search later.)

Reply:
> keep the Advanced version in its own service

@Elleppi , if the new search now basically has almost identical look & feel (with only the request body being different) - maybe having it next to the existing search (with a new name) would also be OK? (In the end it's still "search by twin" - unlike e.g. the meta service which, as you know, is query/update of SPARQL - something different.)

Reply:
Yes, it would be OK :+1: